presto-mysql/src/main/java/io/prestosql/plugin/mysql/ImplementAvgBigint.java

```diff
@@ -0,0 +1,70 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.mysql;
+
+import io.prestosql.matching.Capture;
+import io.prestosql.matching.Captures;
+import io.prestosql.matching.Pattern;
+import io.prestosql.plugin.jdbc.JdbcColumnHandle;
+import io.prestosql.plugin.jdbc.JdbcExpression;
+import io.prestosql.plugin.jdbc.JdbcTypeHandle;
+import io.prestosql.plugin.jdbc.expression.AggregateFunctionRule;
+import io.prestosql.spi.connector.AggregateFunction;
+import io.prestosql.spi.expression.Variable;
+
+import java.sql.Types;
+import java.util.Optional;
+
+import static com.google.common.base.Verify.verify;
+import static com.google.common.base.Verify.verifyNotNull;
+import static io.prestosql.matching.Capture.newCapture;
+import static io.prestosql.plugin.jdbc.expression.AggregateFunctionPatterns.basicAggregation;
+import static io.prestosql.plugin.jdbc.expression.AggregateFunctionPatterns.expressionType;
+import static io.prestosql.plugin.jdbc.expression.AggregateFunctionPatterns.functionName;
+import static io.prestosql.plugin.jdbc.expression.AggregateFunctionPatterns.singleInput;
+import static io.prestosql.plugin.jdbc.expression.AggregateFunctionPatterns.variable;
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static io.prestosql.spi.type.DoubleType.DOUBLE;
+import static io.prestosql.spi.type.IntegerType.INTEGER;
+import static java.lang.String.format;
+
+public class ImplementAvgBigint
+        implements AggregateFunctionRule
+{
+    private static final Capture<Variable> INPUT = newCapture();
+
+    @Override
+    public Pattern<AggregateFunction> getPattern()
+    {
+        return basicAggregation()
+                .with(functionName().equalTo("avg"))
+                .with(singleInput().matching(
+                        variable()
+                                .with(expressionType().matching(type -> type == BIGINT || type == INTEGER))
+                                .capturedAs(INPUT)));
+    }
+
+    @Override
+    public Optional<JdbcExpression> rewrite(AggregateFunction aggregateFunction, Captures captures, RewriteContext context)
+    {
+        Variable input = captures.get(INPUT);
+        JdbcColumnHandle columnHandle = (JdbcColumnHandle) context.getAssignments().get(input.getName());
+        verifyNotNull(columnHandle, "Unbound variable: %s", input);
+        verify(aggregateFunction.getOutputType() == DOUBLE);
+
+        return Optional.of(new JdbcExpression(
+                format("avg(CAST(%s AS double))", columnHandle.toSqlExpression(context.getIdentifierQuote())),
```

Comment:
Could you please fix the CI failure?
```
2020-07-24T00:55:02.4084660Z [ERROR]   TestMySqlDistributedQueries>AbstractTestQueries.testExistsSubquery:1025->AbstractTestQueryFramework.assertQuery:120->AbstractTestQueryFramework.assertQuery:125 Execution of 'actual' query failed: SELECT o1.orderkey, COUNT(*) FROM orders o1  JOIN (SELECT * FROM orders LIMIT 10) o2 ON (SELECT avg(orderkey) FROM orders) > 3 GROUP BY o1.orderkey ORDER BY o1.orderkey LIMIT 5
2020-07-24T00:55:02.4086032Z [ERROR]   TestMySqlDistributedQueries>AbstractTestQueries.testScalarSubquery:907->AbstractTestQueryFramework.assertQuery:120->AbstractTestQueryFramework.assertQuery:125 Execution of 'actual' query failed: SELECT DISTINCT orderkey FROM lineitem WHERE orderkey BETWEEN   (SELECT avg(orderkey) FROM orders) - 10    AND   (SELECT avg(orderkey) FROM orders) + 10
2020-07-24T00:55:02.4086880Z [ERROR]   TestMySqlIntegrationSmokeTest>AbstractTestIntegrationSmokeTest.testAggregation:104->AbstractTestQueryFramework.assertQuery:130 Execution of 'actual' query failed: SELECT avg(nationkey) FROM nation
2020-07-24T00:55:02.4087788Z [ERROR]   TestMySqlIntegrationSmokeTest.testAggregationPushdown:242->AbstractTestIntegrationSmokeTest.assertAggregationPushedDown:452->AbstractTestQueryFramework.computeActual:105->AbstractTestQueryFramework.computeActual:110 » Runtime
```

```
mysql> SELECT cast(1 AS DOUBLE);
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'DOUBLE)' at line 1
```

Reply:
Yeah, I'm investigating the cause of it.

Reply:
@ebyhr Just a question. what version of mysql did you use to reproduce it?
I can't reproduce it in mysql 8.0.
 
```
mysql> SELECT cast(1 AS DOUBLE);
+-------------------+
| cast(1 AS DOUBLE) |
+-------------------+
|                 1 |
+-------------------+
1 row in set (0.00 sec)

mysql> select version();
+-----------+
| version() |
+-----------+
| 8.0.19    |
+-----------+
1 row in set (0.00 sec)
```

Reply:
We are using version 8.0.12.
https://github.com/prestosql/presto/blob/337dd30d26a0ace0b5b2bd99ccedcc0a556f35de/presto-mysql/src/test/java/io/prestosql/plugin/mysql/TestingMySqlServer.java#L30

Reply:
I see. And you are right.
> DOUBLE
Produces a DOUBLE result. Added in MySQL 8.0.17.

https://dev.mysql.com/doc/refman/8.0/en/cast-functions.html#function_cast